src/main/java/org/benetech/servicenet/adapter/shared/util/LocationUtils.java

```diff
@@ -0,0 +1,28 @@
+package org.benetech.servicenet.adapter.shared.util;
+
+import org.benetech.servicenet.adapter.shared.model.Coordinates;
+
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+public final class LocationUtils {
+
+    public static Optional<Coordinates> getCoordinatesFromString(String data, String delimiter) {
+        try {
+            Coordinates result = new Coordinates();
+            String[] parts = data.split(delimiter);
+            result.setLatitude(Double.valueOf(parts[0]));
+            result.setLongitude(Double.valueOf(parts[1]));
+            return Optional.of(result);
+        } catch (NumberFormatException | IndexOutOfBoundsException ex) {
```

Comment:
We don't really want to swallow exception like this, we should rather use conditional constructs to make sure parsing is correct